### ISSUE TITLE
Footer logo links focus indicator missing

### ISSUE THREAD

**Issue by GCHQ-Developer-847** [mi6/ic-ui-kit#2190]
## Summary of the bug 
In our storybook, in Chrome, the logo links on ICDS footers do not have focus indicators, when they're using 'logo' slot.
On Safari, these logo links on ICDS footers show an underline like normal links when focussed. 

TL;DR: Enforce focus indicator on slot='logo' footer links, in line with other footer links.

## 🪜 How to reproduce 
1. Go to the ['With logo links' footer story](https://mi6.github.io/ic-ui-kit/branches/develop/web-components/?path=/story/web-components-footer--with-logo-links) on Safari.
2. Hover on or tab to the logo links.
3. See that they show a line underneath similar to the normal footer links.

I don't think this is correct because we only use that focus state elsewhere on links that are just text.

## 📸 Screenshots or code 
<img src="https://github.com/user-attachments/assets/22e45972-d688-42e2-927a-e0a81f6c834c" width="200">

## 🖥 📱 Device 
- Browser: Safari 

## 🧐 Expected behaviour 
There are few things to do / think about here:
1. Stop the underline from appearing in Safari on hover and focus.
2. In Chrome, there is no focus indicator at all...
- I think this is mainly due to the fact that the developer slots their own links / images in. Should we set the ICDS focus indicator on the link they slot in?
- ^ On the website, we apply the ICDS focus indicator on the slotted links, rather than using slot='logo'
3. Question for a designer - should we add a hover state to logo links? How would this work or look?

**Comment by GCHQ-Developer-847** [comment]
Added the "ICDS Core Team" label because of the need to discuss the hover state

**Comment by GCHQ-Developer-847** [comment]
The logo links no longer show the underline in Safari - they just don't have a focus indicator at all, like in Chrome
